rllib/core/models/tf/encoder.py

```diff
@@ -97,18 +95,14 @@ def get_output_specs(self) -> Optional[Spec]:
                 ENCODER_OUT: TensorSpec(
                     "b, d", d=self.config.output_dims[0], framework="tf2"
                 ),
-                STATE_OUT: None,
             }
         )
 
     @override(Model)
-    def _forward(self, inputs: NestedDict, **kwargs) -> NestedDict:
-        return NestedDict(
-            {
-                ENCODER_OUT: self.net(inputs[SampleBatch.OBS]),
-                STATE_OUT: inputs[STATE_IN],
-            }
-        )
+    def _forward(self, inputs: dict, **kwargs) -> dict:
+        x = inputs
```

Comment:
I think it's un-expected that inputs would change when they go through the forward methods. We should revert this behavior?